virttest/utils_net.py

```diff
@@ -3193,34 +3198,55 @@ def gen_ipv4_addr(network_num="10.0.0.0", network_prefix="24", exclude_ips=[]):
             yield str(ip_address)
 
 
-def get_ip_address_by_interface(ifname, ip_ver="ipv4"):
+def get_ip_address_by_interface(ifname, ip_ver="ipv4", linklocal=False):
     """
     returns ip address by interface
     :param ifname: interface name
     :param ip_ver: Host IP version, ipv4 or ipv6.
+    :param linklocal: Whether ip address is local or remote
     :raise NetError: When failed to fetch IP address (ioctl raised IOError.).
     """
     if ip_ver == "ipv6":
         ver = netifaces.AF_INET6
+        linklocal_prefix = "fe80"
     else:
         ver = netifaces.AF_INET
+        linklocal_prefix = "169.254"
     addr = netifaces.ifaddresses(ifname).get(ver)
+
     if addr is not None:
-        return [a['addr'] for a in addr if not a['addr'].startswith('fe80')][0]
+        if linklocal:
+            try:
+                return [a['addr'] for a in addr
+                        if a['addr'].lower().startswith(linklocal_prefix)][0]
+            except IndexError:
+                logging.warning("No Linklocal IP address configured "
+                                "for the network interface "
+                                "%s !", ifname)
+                return None
+        else:
+            try:
+                return [a['addr'] for a in addr
+                        if not a['addr'].lower().startswith(linklocal_prefix)][0]
+            except IndexError:
+                logging.warning("No IP address configured for "
+                                "the network interface %s !", ifname)
+                return None
```

Comment:
you can handle 3218-3234 in the same way as you have handled in 1441-1449. it helps to reduce no of lines.